internal/configs/version1/nginx-plus.tmpl

```diff
@@ -113,6 +129,23 @@ http {
     include oidc/oidc_common.conf;
     {{- end}}
 
+    {{- if .AppProtectDosLoadModule}}
+        {{- if .AppProtectDosLivenessEnable}}
```

Comment:
If want to add need manually add to the deployment the args inside ports:
        - name: liveness-port
          containerPort: 8090
 livenessProbe:
          httpGet:
            path: app_protect_dos_liveness
            port: liveness-port
          initialDelaySeconds: 60
          periodSeconds: 10

Also add to config map:
app-protect-dos-liveness-enable: "True"

I saw that there is an implementation of readiness probe but not for liveness, so I add the implementation for liveness that we add in the app-protect-dos moudle to KIC

Reply:
Yep, currently we don't have a liveness probe (my bad, I mentioned the readiness probe instead of liveness)

However, the IC process monitors NGINX master process and the DOS process. If they crash, the IC will terminate. I wonder if it is enough?

Ultimately, if we want to have a liveness probe, it should probe the IC, not DOS component, because the DOS component is only part of the story. So the IC needs to be responsible for monitoring the processes it starts.

Also, one potential problem I see with monitoring liveness via an HTTP request to NGINX, is that if NGINX is under pressure (let's say a lot of RPS from clients), it might temporarily not be able to respond. If you consider that typically there are multiple IC pods running and let's say Kubernetes restarts one IC pod because of the failed liveness probe (which is based on making an HTTP request to NGINX), the other pods, which are also probably already under pressure, will immediately get the share of the killed pod's traffic and as a result, they might get killed as well...

Reply:
However, the IC process monitors NGINX master process and the DOS process. If they crash, the IC will terminate. I wonder if it is enough?
There can be an option of admd deadlock for example ,so, the liveness probe will find it.
I agree that it should probe IC, but for now until liveness will be to IC there is an option to use dos liveness.
potential problem - maybe need to add threshold for the liveness probe?